handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java

```diff
@@ -1293,10 +1313,29 @@ public final synchronized SSLEngineResult unwrap(ByteBuffer src, ByteBuffer[] ds
     }
 
     @Override
-    public final Runnable getDelegatedTask() {
-        // Currently, we do not delegate SSL computation tasks
-        // TODO: in the future, possibly create tasks to do encrypt / decrypt async
-        return null;
+    public final synchronized Runnable getDelegatedTask() {
+        if (isDestroyed()) {
+            return null;
+        }
+        final Runnable task = SSL.getTask(ssl);
+        if (task == null) {
+            return task;
```

Comment:
why not just return null?

Reply:
(by returning task it looks like a typo with the equals check)

Reply:
+1 to returning null. I noticed it, but let it slide. But it slowed me down as well.